From 7f84a6cb9ea9abc8fa9bbb1eccb5d7d297a86536 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20K=C3=B6ppeler?= <j.koeppeler@tu-berlin.de>
Date: Tue, 10 Mar 2026 14:23:04 +0100
Subject: [PATCH] net/sched: sch_cake: increase sync time for testing purposes
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Jonas Köppeler <j.koeppeler@tu-berlin.de>
---
 net/sched/sch_cake.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/sched/sch_cake.c b/net/sched/sch_cake.c
index dd4e4ff503a5..22db74a593fd 100644
--- a/net/sched/sch_cake.c
+++ b/net/sched/sch_cake.c
@@ -2811,7 +2811,7 @@ static void cake_config_init(struct cake_sched_config *q, bool is_shared)
 			       */
 	q->rate_flags |= CAKE_FLAG_SPLIT_GSO;
 	q->is_shared = is_shared;
-	q->sync_time = 200 * NSEC_PER_USEC;
+	q->sync_time = 800 * NSEC_PER_USEC;
 }
 
 static int cake_init(struct Qdisc *sch, struct nlattr *opt,
-- 
2.39.5

