--- a/drivers/net/wireless/ath/ath11k/mac.c
+++ b/drivers/net/wireless/ath/ath11k/mac.c
@@ -5294,45 +5294,6 @@ exit:
 	return ret;
 }
 
-static int ath11k_mac_op_conf_tx_mu_edca(struct ieee80211_hw *hw,
-					 struct ieee80211_vif *vif,
-					 unsigned int link_id, u16 ac,
-					 const struct ieee80211_tx_queue_params *params)
-{
-	struct ath11k_vif *arvif = ath11k_vif_to_arvif(vif);
-	struct ath11k *ar = hw->priv;
-	struct wmi_wmm_params_arg *p;
-	int ret;
-
-	switch (ac) {
-	case IEEE80211_AC_VO:
-		p = &arvif->muedca_params.ac_vo;
-		break;
-	case IEEE80211_AC_VI:
-		p = &arvif->muedca_params.ac_vi;
-		break;
-	case IEEE80211_AC_BE:
-		p = &arvif->muedca_params.ac_be;
-		break;
-	case IEEE80211_AC_BK:
-		p = &arvif->muedca_params.ac_bk;
-		break;
-	default:
-		ath11k_warn(ar->ab, "error ac: %d", ac);
-		return -EINVAL;
-	}
-
-	p->cwmin = u8_get_bits(params->mu_edca_param_rec.ecw_min_max, GENMASK(3, 0));
-	p->cwmax = u8_get_bits(params->mu_edca_param_rec.ecw_min_max, GENMASK(7, 4));
-	p->aifs = u8_get_bits(params->mu_edca_param_rec.aifsn, GENMASK(3, 0));
-	p->txop = params->mu_edca_param_rec.mu_edca_timer;
-
-	ret = ath11k_wmi_send_wmm_update_cmd_tlv(ar, arvif->vdev_id,
-						 &arvif->muedca_params,
-						 WMI_WMM_PARAM_TYPE_11AX_MU_EDCA);
-	return ret;
-}
-
 static int ath11k_mac_op_conf_tx(struct ieee80211_hw *hw,
 				 struct ieee80211_vif *vif,
 				 unsigned int link_id, u16 ac,
@@ -5378,15 +5339,6 @@ static int ath11k_mac_op_conf_tx(struct
 		goto exit;
 	}
 
-	if (params->mu_edca) {
-		ret = ath11k_mac_op_conf_tx_mu_edca(hw, vif, link_id, ac,
-						    params);
-		if (ret) {
-			ath11k_warn(ar->ab, "failed to set mu_edca params: %d\n", ret);
-			goto exit;
-		}
-	}
-
 	ret = ath11k_conf_tx_uapsd(ar, vif, ac, params->uapsd);
 
 	if (ret)
